src/app/common-components/header/header.component.ts

```diff
@@ -10,19 +13,18 @@ import { Router } from "@angular/router";
 export class HeaderComponent implements OnInit {
     isAuthenticated = false;
     userName: string;
-    constructor(private authService: AuthService, private router: Router) {}
+    constructor(private store: Store<AppState>) {}
 
     ngOnInit() {
-        this.authService.isAuthenticated$.subscribe(isAuthenticated => {
-            this.isAuthenticated = isAuthenticated;
-        });
-        this.authService.user$.subscribe(user => {
-            this.userName = user.name;
-        });
+        combineLatest([this.store.select(userNameSelector), this.store.select(isUserAuthenticatedSelector)]).subscribe(
```

Comment:
Also, facade pattern can be used for store operations.
core/store-facades/users-store-facade.service.ts
it is a service which has public methods and properties. So, you should use store.select and store.dispatch only in store-facade.